Automattic/wp-calypso
# Domain registrant verification landing page

#### Changes proposed in this Pull Request

In order to provide a more consistent and controllable user experience around domains messaging, we are implementing our own domains messaging. Along with that, we also need to provide some simple landing pages for a variety of user interactions such as verifying domain contact information.

In this PR, we want to add the first of these landing pages as well as a framework that will allow us to easily add more domain-messaging related landing pages.

Depends on D26351-code

A successful verification would look like the following:

![2019-04-05 13 05 16](https://user-images.githubusercontent.com/1379730/55645265-20eb7400-57a6-11e9-88f8-a1f346327aff.gif)

#### Testing instructions

Apply the patch to the back end and build this branch locally.

Visit: http://calypso.localhost:3000/domain-services/registrant-verification?domain=transfer-cancel-test-2.com&email=5test2@test.com&token=bf772d0c0cf7902917d367d124c74682

This will likely show an error because the contact info for this domain is not in a state where it can be re-verified.

To test the various page states, hack the `/verify-email` endpoint from D26351-code to return each of the following:

`return [ 'success' => true ];` to see the success case:

<img width="797" alt="Screen Shot 2019-04-05 at 1 07 01 PM" src="https://user-images.githubusercontent.com/1379730/55645292-32348080-57a6-11e9-8958-1df2746e041a.png">

`return new WP_Error( 'whois_error', 'There was a problem processing your request, please try again later.', 500 );`
-or-
`return new WP_Error( 'bad_request', 'This is not a proper request.', 400 );` should show a generic error page:

<img width="797" alt="Screen Shot 2019-04-05 at 1 07 35 PM" src="https://user-images.githubusercontent.com/1379730/55645317-424c6000-57a6-11e9-8695-7d0ca7ddcf6d.png">


`return new WP_Error( 'unauthorized', 'You are not allowed to execute this action.', 403 );` indicates that the token has expired and should show an error with an offer to re-send the email with a valid token:

<img width="797" alt="Screen Shot 2019-04-05 at 1 07 54 PM" src="https://user-images.githubusercontent.com/1379730/55645364-60b25b80-57a6-11e9-8884-86201f1b4358.png">


`return new WP_Error( 'email_mismatch', 'The email you are trying to verify is different from the email we have on record.', 500 );` Should show an error page with some information about the reason for the error:

<img width="797" alt="Screen Shot 2019-04-05 at 1 26 48 PM" src="https://user-images.githubusercontent.com/1379730/55645422-893a5580-57a6-11e9-9199-15d3b0496780.png">


Additionally, try visiting the path with an invalid `/:action` (ex.: http://calypso.localhost:3000/domain-services/registrant-verificationzzz?domain=transfer-cancel-test-2.com&email=5test2@test.com&token=bf772d0c0cf7902917d367d124c74682) to make sure you see the equivalent of a 404 page for these domain services landing pages:

<img width="797" alt="Screen Shot 2019-04-05 at 1 08 56 PM" src="https://user-images.githubusercontent.com/1379730/55645436-93f4ea80-57a6-11e9-966c-5ff83770c975.png">
